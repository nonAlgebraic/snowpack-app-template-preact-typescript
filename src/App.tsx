import { h, FunctionComponent } from 'preact';
import { useEffect } from 'preact/hooks';
import * as styles from './App-style';
import logo from './logo.png';

const App: FunctionComponent = () => {
  useEffect(() => {
    console.log('foo');
  }, []);
  return (
    <div className={styles.container}>
      <header className={styles.header}>
        <img src={logo} className={styles.logo} alt="logo" />
        <p>
          Edit <code>src/App.tsx</code> and save to reload.
        </p>
        <a
          className={styles.link}
          href="https://preactjs.com"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn Preact
        </a>
      </header>
    </div>
  );
};

export default App;
