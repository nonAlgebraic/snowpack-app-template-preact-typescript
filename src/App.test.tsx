import { h } from 'preact';
import { render } from '@testing-library/preact';
import { axe } from 'jest-axe';
import App from './App';

describe('App.tsx', () => {
  it('renders correctly', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('renders learn preact link', () => {
    const { getByText } = render(<App />);
    const linkElement = getByText(/learn preact/i);
    expect(linkElement).toBeInTheDocument();
  });

  it('has no accessibility errors', async () => {
    const { container } = render(<App />);

    expect(await axe(container)).toHaveNoViolations();
  });
});
