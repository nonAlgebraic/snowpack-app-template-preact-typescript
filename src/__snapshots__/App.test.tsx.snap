// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.tsx renders correctly 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  to {
    -webkit-transform: scale(1.06);
    -ms-transform: scale(1.06);
    transform: scale(1.06);
  }
}

.emotion-3 {
  text-align: center;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 100vh;
  color: white;
  font-size: calc(10px + 2vmin);
  background-color: #282c34;
}

.emotion-1 {
  color: #61dafb;
}

.emotion-0 {
  height: 40vmin;
  margin-bottom: 1rem;
  -webkit-animation: animation-0 infinite 1.6s ease-in-out alternate;
  animation: animation-0 infinite 1.6s ease-in-out alternate;
  pointer-events: none;
}

<div>
  <div
    class="emotion-3"
  >
    <header
      class="emotion-2"
    >
      <img
        alt="logo"
        class="emotion-0"
        src="logo.png"
      />
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to reload.
      </p>
      <a
        class="emotion-1"
        href="https://preactjs.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn Preact
      </a>
    </header>
  </div>
</div>
`;
